go/client/logclient.go

```diff
@@ -126,6 +136,66 @@ func (c *LogClient) fetchAndParse(uri string, res interface{}) error {
 	return nil
 }
 
+// Makes a HTTP POST call to |uri|, and attempts to parse the response as a JSON
+// representation of the structure in |res|.
+// Returns a non-nil |error| if there was a problem.
+func (c *LogClient) postAndParse(uri string, req interface{}, res interface{}) error {
+	post_body, err := json.Marshal(req)
+	if err != nil {
+		return err
+	}
+	http_req, _ := http.NewRequest("POST", uri, bytes.NewReader(post_body))
+	resp, err := c.httpClient.Do(http_req)
```

Comment:
Why not just this (or something close to it, my Go-fu isn't that strong)?

    resp, err := c.httpClient.Post(uri, "application/json", post_body)